Source/ASImageNode.mm

```diff
@@ -483,53 +480,45 @@ + (UIImage *)createContentsForkey:(ASImageNodeContentsKey *)key drawParameters:(
 
   // Use contentsScale of 1.0 and do the contentsScale handling in boundsSizeInPixels so ASCroppedImageBackingSizeAndDrawRectInBounds
   // will do its rounding on pixel instead of point boundaries
-  ASGraphicsBeginImageContextWithOptions(contextBounds.size, contextIsOpaque, 1.0);
-  
-  BOOL contextIsClean = YES;
-  
-  CGContextRef context = UIGraphicsGetCurrentContext();
-  if (context && key.willDisplayNodeContentWithRenderingContext) {
-    key.willDisplayNodeContentWithRenderingContext(context, drawParameters);
-    contextIsClean = NO;
-  }
-  
-  // if view is opaque, fill the context with background color
-  if (key.isOpaque && key.backgroundColor) {
-    [key.backgroundColor setFill];
-    UIRectFill({ .size = key.backingSize });
-    contextIsClean = NO;
-  }
-  
-  // iOS 9 appears to contain a thread safety regression when drawing the same CGImageRef on
-  // multiple threads concurrently.  In fact, instead of crashing, it appears to deadlock.
-  // The issue is present in Mac OS X El Capitan and has been seen hanging Pro apps like Adobe Premiere,
-  // as well as iOS games, and a small number of ASDK apps that provide the same image reference
-  // to many separate ASImageNodes.  A workaround is to set .displaysAsynchronously = NO for the nodes
-  // that may get the same pointer for a given UI asset image, etc.
-  // FIXME: We should replace @synchronized here, probably using a global, locked NSMutableSet, and
-  // only if the object already exists in the set we should create a semaphore to signal waiting threads
-  // upon removal of the object from the set when the operation completes.
-  // Another option is to have ASDisplayNode+AsyncDisplay coordinate these cases, and share the decoded buffer.
-  // Details tracked in https://github.com/facebook/AsyncDisplayKit/issues/1068
-
-  BOOL canUseCopy = (contextIsClean || imageIsOpaque);
-  CGBlendMode blendMode = canUseCopy ? kCGBlendModeCopy : kCGBlendModeNormal;
-  
-  @synchronized(image) {
-    [image drawInRect:imageDrawRect blendMode:blendMode alpha:1];
-  }
-  
-  if (context && key.didDisplayNodeContentWithRenderingContext) {
-    key.didDisplayNodeContentWithRenderingContext(context, drawParameters);
-  }
+  UIImage *result = ASGraphicsCreateImageWithOptions(key.backingSize, key.isOpaque, 1.0, image, ^{
```

Comment:
This previously would not choose opaque if the image wouldn't fill the context (now a new UIImage, but apparently sized the same). Is this change intentional?

Reply:
Oh good call, this was a bad merge. Fixed